x-pack/plugins/cloud_security_posture/public/pages/findings/findings_flyout/overview_tab.tsx

```diff
@@ -99,6 +118,40 @@ const getEvidenceList = ({ result }: CspFinding) =>
   ].filter(Boolean) as EuiDescriptionListProps['listItems'];
 
 export const OverviewTab = ({ data }: { data: CspFinding }) => {
+  const {
+    services: { discover },
+    notifications: { toasts },
+  } = useKibana();
+  const latestFindingsDataView = useLatestFindingsDataView();
+
+  const navigateToIndex = useCallback(async () => {
+    try {
+      // both cases should not happen, data view is loaded beforehand on findings page, this is mainly to discriminate and as a precaution
+      if (!discover.locator || !latestFindingsDataView.data) {
+        throw new Error();
+      }
```

Comment:
maybe add a message like `missing locator or data view` 

although we might avoid the need to throw if we don't render a link in these cases

Reply:
I used to have a more descriptive error message, @einatjacoby consulted with the technical writing team and in the end, it was decided to only display the same message always (`index link not found`)

Reply:
it's a bit odd to throw an error without a message and when caught to show some message without looking at the `err` 

could we call the toasts here and avoid catching at all?

Reply:
i left the catch so if `navigate` will throw for some reason we still get the toast

Reply:
but now it throws the same error for 2 types if possible errors. 

we can instead `return` here after we've shown an danger toast

and if `navigate()` throws, we could pipe the catched error to `toasts.addError`

this way each type of error is handled on its own and users will see correct message in each toast

Reply:
if this needs further explanation lets take it offline. as mentioned, specific case errors are not wanted. this is a ux decision and not code decision. i was asked not to display a navigate error which can be any different than "Index link not found". 
in all of the above cases, navigation throw, or missing deps, we want to show the exact same toast with the same text. code implementation makes sure any error inside this function throws into the same branch which leads to the same result which is the same toast with the same text, as requested.